### ISSUE TITLE
Failed to create document from template

### ISSUE THREAD

**Issue by R7JANV1** [nextcloud/android#10594]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. create a new text document
2. see the following error message
![Screenshot_2022-08-04-23-22-46-32_06ea656a9ba653ff96746f477f32d0f9](https://user-images.githubusercontent.com/90777102/182955539-29b12673-7519-46e5-a8f9-c9aa8f658057.jpg)

### Expected behaviour

a text document should be created

### Actual behaviour

no text document is created

### Android version

12

### Device brand and model

Realme RMX3085

### Stock or custom OS?

Stock

### Nextcloud android app version

3.20.3

### Nextcloud server version

24.0.3

### Using a reverse proxy?

No

**Comment by AlvaroBrey** [comment]
Thanks for your report. Can you get a logcat from the Android device so we can see if there's any errors? Instructions here: https://github.com/nextcloud/android#getting-debug-info-via-logcat-mag

Feel free to search and replace the server URL, username and other file names in the output.

**Comment by R7JANV1** [comment]
It may be related to this in the server config 'skeletondirectory' => '',

**Comment by R7JANV1** [comment]
> Thanks for your report. Can you get a logcat from the Android device so we can see if there's any errors? Instructions here: https://github.com/nextcloud/android#getting-debug-info-via-logcat-mag
> 
> Feel free to search and replace the server URL, username and other file names in the output.

no unfortunately not and the link does not work

**Comment by R7JANV1** [comment]
Can be reproduced as follows:
1. install Nextcloud AIO Server (stable)
2. config.php customization
     'skeletondirectory' => '',
     'knowledgebaseenabled' => false,
3. connect Nextcloud Android (or IOS) app with account
4. Create new document
5. see the error message

**Comment by AlvaroBrey** [comment]
That's interesting. Empty `skeletondirectory` should prevent templates from being copied on user creation, but apart from that should not make anything crash. Will investigate

**Comment by AlvaroBrey** [comment]
Couldn't reproduce even with the above steps. Will have to try again

**Comment by DerLinkman** [comment]
I think i have a similar problem on iOS 16 as well.

If i want to create a new .odt file within the Nextcloud App on iOS i get this request in the logs: (the 212.50 is my caddy reverse proxy btw):

`192.168.212.50 - DerLinkman [20/Sep/2022:17:10:15 +0000] "POST /ocs/v2.php/apps/richdocuments/api/v1/templates/new?format=json HTTP/1.1" 500 111 "-" "Mozilla/5.0 (iOS) Nextcloud-iOS/4.4.4" "CENSORED_IPV6"`

It results in a Error 500 (as seen in the response code of the request above).

Interestingly if i open a existing Document it works like a charm without an error.

**Comment by R7JANV1** [comment]
Yes, this is also the behavior on Android.

**Comment by R7JANV1** [comment]
> Couldn't reproduce even with the above steps. Will have to try again

Have you made any progress?

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky this is the issue I am also experiencing with our production server and `master` branch of the Android client

**Comment by R7JANV1** [comment]
I can reproduce it. The error is present on all my Nextcloud instances

**Comment by juliushaertl** [comment]
Could you check your Nextcloud log if there is an error similar to this one or a different one?

`Undefined array key \"dirname\" at /var/www/nextcloud/apps/richdocuments/lib/Controller/OCSController.php#322`

If it is the same error then creating documents should only fail in the root directory and a fix would be in https://github.com/nextcloud/richdocuments/pull/2543

**Comment by R7JANV1** [comment]
Yes this is the same error.

**Comment by juliushaertl** [comment]
Fix was merged and back ported and will become available with the next maintenance releases of richdocuments.

**Comment by anoegit** [comment]
I had the same problem. When creating new Office or text files, the error message always appeared: "Failed to create document from template". In addition, you couldn't use the "save as" function in Collabora.
Since I couldn't narrow down the error even after 2 months, I searched my instance for changes compared to an instance from 3 months ago. Then I deactivated some apps that had been installed since then. When I deactivated the "Analytics" app, everything worked properly again.
The "Analytics" app by author Marcel Scherello is definitely the cause of the problem.
